src/scene/batching.js

```diff
@@ -0,0 +1,719 @@
+                            }
+                        }
+                    }
+                    if (paramFailed) {
+                        meshInstancesLeft2.push(meshInstancesLeft[i]);
+                        continue;
+                    }
+                    // Split by static/non static
+                    params2 = meshInstancesLeft[i]._staticLightList;
+                    if ((lightList && !params2) || (!lightList && params2)) {
+                        meshInstancesLeft2.push(meshInstancesLeft[i]);
+                        continue;
+                    }
+                    // Split by static light list
+                    if (lightList && params2) {
+                        paramFailed = false;
+                        for(k=0; k<lightList.length; k++) {
+                            if (params2.indexOf(lightList[k]) < 0) {
+                                paramFailed = true;
+                                break;
+                            }
+                        }
+                        for(k=0; k<params2.length; k++) {
+                            if (lightList.indexOf(params2[k]) < 0) {
+                                paramFailed = true;
+                                break;
+                            }
+                        }
+                        if (paramFailed) {
+                            meshInstancesLeft2.push(meshInstancesLeft[i]);
+                            continue;
+                        }
+                    }
+                }
+
+                aabb.add(meshInstancesLeft[i].aabb);
+                vertCount += meshInstancesLeft[i].mesh.vertexBuffer.getNumVertices();
+                lists[j].push(meshInstancesLeft[i]);
+                
+                // Split by instance number
+                if (isDynamic && lists[j].length === maxInstanceCount) {
+                    if (i === meshInstancesLeft.length) {
+                        meshInstancesLeft2 = [];
+                    } else {
+                        meshInstancesLeft2 = meshInstancesLeft.slice(i + 1);
+                    }
+                    break;
+                }
+            }
+            j++;
+            meshInstancesLeft = meshInstancesLeft2;
+        }
+        
+        return lists;
+    };
+
+    /**
+     * @function
+     * @name pc.Batching#create
+     * @description Takes a mesh instance list that has been prepared by prepare(), and returns a pc.Batch object. This method assumes that all mesh instances provided can be rendered in a single draw call.
+     */
+    Batching.prototype.create = function(meshInstances, isDynamic) {
+
+        if (!this._init) {
+            var boneLimit = "#define BONE_LIMIT " + this.device.getBoneLimit() + "\n";
+            this.transformVS = boneLimit + pc.shaderChunks.transformBatchSkinnedVS;
+            this.skinTexVS = pc.shaderChunks.skinTexVS.replace("attribute vec4 vertex_boneIndices", "attribute float vertex_boneIndices");
```

Comment:
This seems fragile. We could break this is someone updates the skin chunk. Should these just be new chunks?